llvm/test/CodeGen/AMDGPU/lower-buffer-fat-pointers-constants.ll

```diff
@@ -176,7 +179,7 @@ define i64 @ptrtoint_short() {
 define i32 @ptrtoint_very_short() {
 ; CHECK-LABEL: define i32 @ptrtoint_very_short
 ; CHECK-SAME: () #[[ATTR0]] {
-; CHECK-NEXT:    ret i32 add nuw nsw (i32 shl (i32 ptrtoint (ptr addrspace(8) @buf to i32), i32 32), i32 12)
+; CHECK-NEXT:    ret i32 12
```

Comment:
This lost depending on the global at all?

Reply:
This is the same as https://github.com/llvm/llvm-project/pull/95543, but for constants. It previously produced a poison shift that shouldn't be there.